### ISSUE TITLE
Use CTest to run unit tests with mpi

### ISSUE THREAD

**Issue by bruno-at-orange** [KhiopsML/khiops#23]
We can setup mpi run in the CMakeLists like this:
```cmake
add_test(NAME parallel-mpi COMMAND ${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${MPIEXEC_MAX_NUMPROCS}
                                   ${MPIEXEC_PREFLAGS} $<TARGET_FILE:parallel_mpi_test> ${MPIEXEC_POSTFLAGS})

# Add "mpi" label to the test and set a default process number to launch.
set_tests_properties(parallel-mpi PROPERTIES LABELS "mpi" PROCESSORS ${MPIEXEC_MAX_NUMPROCS})
```

The parallel run is launch with:
```bash
# run all test labeled by `mpi
ctest --preset linux-gcc-release -L mpi   --output-junit output-mpi.xml
```
All the other tests can be launched with one line:
```bash
ctest --preset linux-gcc-release -LE mpi   --output-junit output-serial.xml
```

It is interesting because it is independent of the mpi implementation (mpich, openmpi etc...) and use the number of processors detected on the host. 

Note: it's tempting to use the `--parallel` flag to execute serial tests concurrently but Khiops is not thread-safe

**Comment by folmos-at-orange** [comment]
@bruno-at-orange  I think this is fixed at least from PR #35

**Comment by bruno-at-orange** [comment]
done with #203
